### PR TITLE
[DPE-2597] Do not change permissions on existing directories

### PR DESCRIPTION
**Pull request by Batalex**
See https://github.com/canonical/zookeeper-operator/pull/138 for a full test suite

### CODE REVIEW

**Review by marcoppenheimer — APPROVED** [review]
Looks good!

Could I ask you to add two comment lines somewhere in the `install` hook:

1. Explaining which user `584788` is
2. A `FIXME:` saying to switch to `584792` + `_daemon_` after 24.04

Maybe not even a switch, maybe we add them both to a group or something for compatibility.

File: snap/hooks/install [hunk 1]
```diff
@@ -8,14 +16,24 @@ export CONF="${SNAP_DATA}"/etc/$COMPONENT
 export LOGS="${SNAP_COMMON}"/var/log/$COMPONENT
 export DATA="${SNAP_COMMON}"/var/lib/$COMPONENT
 
+if [stat -c '%u' "$DATA" == 0 -o ! -d "$DATA"]; then
+    mkdir -p $DATA
+    mkdir -p $LOGS
+
+    chmod -R 770 "${SNAP_COMMON}"
+    chown -R 584788:root "${SNAP_COMMON}"
+    chmod 770 "${DATA}"
```
Comment (deusebio) [inline snap/hooks/install]: **question** I know we have been using `770` before. I'm just wondering whether for the `data` folder (where Kafka, ZooKeeper) writes, maybe we could lower the permission for the group to read only (i.e. `750`).

**Review by deusebio — APPROVED** [review]
LGTM. 


It is not super critical, but I have a small suggestion/question that may be a small (but more secure) improvements. 

If it is not a straightforward change or there are concerns around this, we can skip it for now, as this would be outside of the scope of the PR

File: snap/hooks/install (hunk 1 above)
Comment (Batalex) [reply snap/hooks/install]: Testing changes on snap is a little tricky, as I do not have permissions to publish branches. I'll keep this PR opened while I work on the Kafka snap and test there

Comment (deusebio) [reply snap/hooks/install]: If this is not blocking other things, I'd give it a go. I suppose I should also work to enable you guys to publish temp branches for testing. Let me double check this with the other managers

Comment (Batalex) [reply snap/hooks/install]: Alright, the CI is green with 750!
https://github.com/canonical/zookeeper-operator/actions/runs/8796703126

**Pull request merged**